File: coding-test/src/lyn/프로그래머스/코딩테스트/day241029/정수_내림차순으로_배치하기/solution.js
function solution(n) {
    const arr = String(n)
        .split('')
        .sort()
        .reverse()
        .join('')
    // 스트링으로 바꿔서 숫자 다 자르기



    return +arr;// +붙여서 숫자로 형변환
}

// 함수 solution은 정수 n을 매개변수로 입력받습니다. n의 각 자릿수를 큰것부터 작은 순으로 정렬한 새로운 정수를 리턴해주세요. 예를들어 n이 118372면 873211을 리턴하면 됩니다.
//
// 제한 조건
// n은 1이상 8000000000 이하인 자연수입니다.

File: coding-test/src/lyn/프로그래머스/코딩테스트/day241029/정수_내림차순으로_배치하기/Solution.java
package lyn.프로그래머스.코딩테스트.day241029.정수_내림차순으로_배치하기;

import java.util.Arrays;
import java.util.Comparator;

public class Solution {
    public long solution(long n) {
        long answer = 0;
        String[] arr = String.valueOf(n).split("");
        Arrays.sort(arr, Comparator.reverseOrder());

        // 정렬된 숫자들 문자열로 붙임
        StringBuilder sb = new StringBuilder();
        for (String s : arr) {
            sb.append(s);
        }

        answer = Long.parseLong(sb.toString());

        return answer;
    }
}
